buildSrc/src/test/resources/plugin-descriptor.properties

```diff
@@ -0,0 +1,45 @@
+# Elasticsearch plugin descriptor file
```

Comment:
Why is a duplicate file being added under test?

Reply:
Since I am using the `getClass().xxxxxx` in the test it is looking into `src/test/resource` for `plugin-descriptor.properties` in order to copy it over to `templates` and then perform `property substitution`. 

See https://github.com/elastic/elasticsearch/pull/39605/files#diff-a9a997ce88a166a144b1881bc9783ce6R66

Reply:
This duplicate is not necessary.  Also, isn't the test trying to load the *output* file, not the template?

Reply:
You were right! The most recent helped me understand how the plugin works and I have tweaked the test as well.